### ISSUE TITLE
Default Jewelcrafting profile isn't well optimized

### ISSUE THREAD

**Issue by gomsnothereman** [Phr33d0m/NW-Profession-Bot#164]
I noticed on doing jewel crafting my guys get up to making wizards necklaces(rank16) and belts of renewal (rank18) and that is where they stop (most are at 18 or 19) however they COULD also be converting the wizards necklaces into knight necklaces (rank17) for 900xp (better than the belts of renewal at 18 oddly) yet they are just letting the wizards necklaces needed for the knight necklaces pile up in their bags...

i am guessing the order for jewel crafting is just skipping that task and i figure that isn't how it should be working when it could be converting the 15-30 of those it has made, instead of starting back at square 1 remaking more :)

**Comment by WloBeb** [comment]
You can always define custom profile...  From the time when the bags are bound, I personally use 'Mass refine' profiles for leveling. Purple assets for most professions are really cheap.

**Comment by gomsnothereman** [comment]
i will ahve to look into customizing the profile more, i had been "watching" because i noticed jewelcrafting kinda flew up to 16 then slowed down and STAYED slow till 20..

i just assumed the profiles had like "EVERY SINGLE" task in them, and then it just worked "backwards" from the top to select the highest level one, but after looking at it, i see even that wouldn't be most efficient, for example that wizards necklace ot knight necklace is 900xp which is the best you can get in xp per hour i think until post 20.. kinda odd..

**Comment by dlebedynskyi** [comment]
I have not really looked at optimizing those before. 
First due to at time of creating there was no bound bags issues. 
If you want you can really look at if to build up better strategy.

**Comment by gomsnothereman** [comment]
i am gonna try to take a look at it, is there a decent way to pull all the different task names out??

I am also on 3 characters trying what wlobeb suggested and setting mass refine for some guys at jewelcraft lvl14..

**Comment by noonereally** [comment]
About 3 ways, 
If you click the task, the right side of the URL address will be task name.
You can inspect the button of the task and it has the name. 
Or in git, in the erpo there is a dir, inside a text file
 https://github.com/Phr33d0m/NW-Profession-Bot/tree/master/extra 
@dlebedynskyi  added most of the tasks if not all of them

**Comment by gomsnothereman** [comment]
Quick fix if yall could throw it in? at line 767 it says
17 : ["Jewelcrafting_Tier3_Neck_Offense_3", "Jewelcrafting_Tier3_Waist_Offense_3", "Jewelcrafting_Tier3_Refine_Basic", "Jewelcrafting_Tier3_Gather_Basic", "Jewelcrafting_Tier2_Gather_Basic", "Jewelcrafting_Tier1_Gather_Basic"],

if the "jewelcrafting_tier3_neck_offense_3" could be changed to "jewelcrafting_tier3_neck_defense_3" that is the name of the task to create the knights necklace.. it actually (in the code) has the net_offense_3 listed under level 16 (line 757) and 17 (line 767) the level 16 one IS called offense_3, that creates the wizards necklace... level 17 it is defense_3 which converts the wizards necklace into a knights necklace (only needs residium to do it nothing else, but that + wizards necklace)

that is why it was "skipping" doing that conversion for 900xp..

as for "optimizing, honeslty he did it the way i would of, the belt tasks only take 1 of each resource for a set amount (ever increasing per hour xp gain as you go up in levels) and the rings take 2 of each resource BUT give the SAME xp as the belt, however the necklaces take THREE of each resource but give about 40% more in xp per hour (or task).. so there is ZERO reason to do the rings (and he didn't) but doing the belts is the easiest on resources, and doing the necklaces as you HAVE the resources is a little more XP efficient but takes more resources..

ALL of those only take 1 residuum but that is normal.. if you ONLY cared about saving residium, you could say doing ONLY the belts is more cost effective, but would not be the fastest way.. the way he has it, is fastest..

so the only issue was a that typo on line 767

PS... I dunno how exactly that code works.. so i dunno if that "jewelcrafting_tier3_neck_defense_3" needs to be added to like line 18,19?? because that is a higher task than the level 16 "jewelcrafting_tier3_neck_offense_3

(basing all the XP stuff off here  ( http://neverwinter.gamepedia.com/Jewelcrafting )

**Comment by NameTakenAgain** [comment]
When we're talking about optimizing task lists. I wrote this on greasyfork last year:

---

After running the script for a while I started wondering why it produced so many items that were not used in the production of future items. Here I have focused on two branches in the requirement trees:

A) Witch's Belt -> Belt of Reformation -> Mage's Belt ->Belt of Replenishment ->Wizard's Belt and
B) Witch's Necklace ->Necklace of reformation -> Mage's Necklace -> Necklace of Replenishment -> Wizard's Necklace -> Knight's Necklace.

B yields better XP/hour but we want to focus on A on level 8, 9 and 14, and 15, after the switches to new raw materials. This way we should be able to use everything we already made when making new items and gain XP quicker.

---

One further thing to keep in mind with Jewelcrafting as well as other professions is that we want to reuse what we have done in earlier stages. So for level X we want to produce a Foo_3 and if we have filled three slots making Foo_3 the next best alternative is probably to make a Foo_2. This means we will often have a Foo_2 ready when it is time to start the next Foo_3 task and we will spend less time on lower-XP tasks.
